src/ts/components/subtitleoverlay.ts

```diff
@@ -73,6 +162,125 @@ export class SubtitleOverlay extends Container<ContainerConfig> {
+    if (!this.subtitleManager.hasCues) {
+      this.addComponent(this.forcedSubtitle);
+      this.updateComponents();
+      this.show();
+    }
+  }
+  removeEnforcedSubtitleLabel() {
+    this.forceSubtitle = false;
+    this.removeComponent(this.forcedSubtitle);
+    this.updateComponents();
+  }
+  // Methods used to define custom styling on subtitles labels
+  setColor(color: string) {
+    let col = ColorUtils.colorFromCss(color);
+    col.a = this.config.fontColor.a;
+    this.config.fontColor = col;
+    this.updateSubtitleLabelCss();
+    this.setItem('fontColor', this.config.fontColor.toCSS());
+  }
+  setBackgroundColor(color: string) {
+    let backgroundColor = ColorUtils.colorFromCss(color);
+    if (! this.config.backgroundColor.a || this.config.backgroundColor.a === 0) {
+      // 25%  opacity at least
+      backgroundColor.a = 0.25;
+    } else {
+      backgroundColor.a = this.config.backgroundColor.a;
+    }
+    this.config.backgroundColor = backgroundColor;
+    this.updateSubtitleLabelCss();
+    this.setItem('backgroundColor', this.config.backgroundColor.toCSS());
+  }
+  setWindowColor(color: string) {
+    let windowColor = ColorUtils.colorFromCss(color);
+    if (! this.config.windowColor.a || this.config.windowColor.a === 0) {
+      // 25%  opacity at least
+      windowColor.a = 0.25;
+    } else {
+      windowColor.a = this.config.windowColor.a;
+    }
+    this.config.windowColor = windowColor;
+    this.getDomElement().css('background', this.config.windowColor.toCSS());
+    this.setItem('windowColor', this.config.windowColor.toCSS());
+  }
+  setFontOpacity(alpha: number) {
+    this.config.fontColor.a = alpha;
+    this.updateSubtitleLabelCss();
+    this.setItem('fontColor', this.config.fontColor.toCSS());
+  }
+  setBackgroundOpacity(alpha: number) {
+    this.config.backgroundColor.a = alpha;
+    this.updateSubtitleLabelCss();
+    this.setItem('backgroundColor', this.config.backgroundColor.toCSS());
+  }
+  setWindowOpacity(alpha: number) {
+    this.config.windowColor.a = alpha;
+    this.getDomElement().css('background', this.config.windowColor.toCSS());
+    this.setItem('windowColor', this.config.windowColor.toCSS());
+  }
+  setFontFamily(fontFamily: string) {
+    this.config.fontFamily = fontFamily;
+    this.setItem('fontFamily', this.config.fontFamily);
+    this.updateSubtitleLabelCss();
+  }
+  setFontStyle(fontStyle: string) {
+    this.config.fontStyle = fontStyle;
+    this.setItem('fontStyle', this.config.fontStyle);
+    this.updateSubtitleLabelCss();
+  }
+  setFontVariant(fontVariant: string) {
+    this.config.fontVariant = fontVariant;
+    this.setItem('fontVariant', this.config.fontVariant);
+    this.updateSubtitleLabelCss();
+  }
+  /**
+   * A helper method to avoid updating the CSS label 3 times in a row
+   * since family, style and variant are normally updated together
+   **/
+  setFont(fontFamily: string, fontStyle: string, fontVariant: string) {
```

Comment:
Call `setFontFamily`,  `setFontStyle`, and `setFontVariant` from here with `update` parameter `false` to prevent duplicate code